compiler/infra/TreeServices.cpp

```diff
@@ -189,7 +194,7 @@ TR_AddressTree::process(TR::Node * aiaddNode, bool onlyConsiderConstAiaddSecondC
 
    TR::ILOpCodes opCodeaiAdd = aiaddNode->getOpCodeValue();
 
-   if (opCodeaiAdd != TR::aiadd && opCodeaiAdd != TR::aladd)
+   if (opCodeaiAdd != TR::aiadd && opCodeaiAdd != TR::aladd && !aiaddNode->isDataAddrPointer())
```

Comment:
Given that this node might not always represent an address addition operation, it might be appropriate to rename the variables involved to avoid confusion.  Perhaps, `aiaddNode` should be `addrCalcNode` or `elementAddrNode` or something like that, and similarly for `opCodeaiAdd`.  Names like `aiaddSecondChild`, for instance, could remain as is, because it really still refers to the second child of an `aiadd`/`aladd` node.